connect-button/src/test/java/com/ifttt/connect/ui/QueueOperationsTest.java

```diff
@@ -0,0 +1,69 @@
+package com.ifttt.connect.ui;
+
+import android.app.Activity;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import com.ifttt.connect.ShadowAnimatorSet;
+import java.io.IOException;
+import java.util.HashMap;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.annotation.Config;
+
+import static com.google.common.truth.Truth.assertThat;
+
+@RunWith(AndroidJUnit4.class)
+@Config(shadows = { ShadowAnimatorSet.class })
+public class QueueOperationsTest {
+
+    private final Activity activity = Robolectric.buildActivity(TestActivity.class).create().get();
+    private AnalyticsManager analyticsManager;
+
+    @Before
+    public void setup() {
+        analyticsManager = AnalyticsManager.getInstance(activity);
+
+        try {
+            analyticsManager.queue.clear();
+        } catch (IOException e) {
+
+        }
+
+        AnalyticsManager.destroy();
+    }
+
+    @Test
+    public void testEnqueueAndRead() {
```

Comment:
I wonder if we really need to test these operations – if the implementation is just calling the corresponding QueueFile methods, these are just testing QueueFile, right?

Reply:
yes, changed them to test for the content read from queue. We are providing synchronization over the `ObjectQueue` operations which are not guaranteed to be synchronous.